functioning xflavor, eigvenvectors, and modules

gitneutrinos opened the pull request.

gitneutrinos committed:
Merge pull request #9 from srichers/master

functioning test case for Sherwood

gitneutrinos committed:
Merge pull request #10 from srichers/master

stuff is more disentagled

gitneutrinos committed:
Update StabilityPackage.wl

gitneutrinos committed:
import check

gitneutrinos committed:
Added in basic gershgorin disc functions

gitneutrinos committed:
GDValue,Box Transform fitting, GD test case in runstuff

added in gershgorin things and fixed an error. Runstuff now runs a GDValue verification test.  Added in box transform fit function (generate intial guesses, generate fits for single radius). Changed kgrid bounds, and ktarget uses total potential.

gitneutrinos committed:
fixed a bug in siPotential

needs to know the number of bins n

gitneutrinos committed:
more bugs

all around siPotential.

gitneutrinos committed:
got rid of potential function

siPotential replaces the need for it

gitneutrinos committed:
hs[[9]] no longer exists

removed

gitneutrinos committed:
Run Template, and package updates

Package Changes:

1) Documentation. ?function_name now returns the arguments required.

2) no dh functions for build hamiltonians, get equations, and kadapt.

3) buildkgrid and kadapt now have optional arguments. A manual k target, or manual bounds on the kgrid can be input. Otherwise, it defaults to automatic targetting and bounds.

4) ellispe box fit. Contains a few functions to generate intial guesses, fit a single radius, and then fit the entire profile. Needs testing/debugging.

5) Fixed GD functions. Needs debugging.

6) makeLocalVariables[] function added to import constants into the run notebook, for conversion purposes ect.

7) Fewer exported functions. Several functions are now missing from the private preamble as they are rarely, if ever, called outside of the package.

8) SI Potential function has been pulled out of the buildHamiltonians function. Returns sqrt[2] Gf (total neutrino number + antis).

Changes to run template:
1) starting to add 2 beam check to the accruacy checks block. Need to figure out the artifical frequency and lotosdo values to appropriately sub in for the mus in the self-interaction.  Trying to avoid defining new buildHamiltonian function if possible.

2) Plotting functions block.  Added some automatic plotting routines.

gitneutrinos committed:
Less Functions

Removed the three "no dh" functions and implemented options instead. Calling kAdapt with the option "dh"-> False removes the hamiltonian perturbation.

gitneutrinos committed:
2 beam checks

2 beam check in runtemplate works. Gets correct answer. Some other clean up of the package.

gitneutrinos committed:
Box fit improvements

Removed the "get intial guess" function. Instead, if the radius entered for the fitsingleradius function is the first radial bin, generates the appropriate guesses at that time.

gitneutrinos committed:
x flavor options

Added options to most functions to allow for non-zero x flavors.  Currently does not actually add them into the Hsi, just finds their potential and calculates starting density matricies.

New option patterns for buildkGrid and kAdapt.

Bug still present in kAdapt if "xflavor"-> True. Works if false. unclear where issue lies, as each function called with xflavor-> true works indvidually.

gitneutrinos committed:
Delete RunStuff.wl

gitneutrinos committed:
Delete RunStuff Template.nb

gitneutrinos committed:
ndensity function & box fit intialguess

Added a function which computes the mu matricies for nu, nubar, and nux neutrino flavors.  This is the only place where the "xflavor" option makes an actual change, as whether mux is zero or nonzero propogates through.  The option is still required on most functions.  Seperated out the intial guess for the box fit into ti's own function.

gitneutrinos committed:
Update StabilityPackage.wl

gitneutrinos committed:
Modules, eigenvectors, xflavor debug,  xflavor ellipse fit

Bug issue due to the static nature of block variables. All blocks are now modules, which are more flexible. Slight slow down is possible.

Eigenvectors can now be found with evecscale function.

New option in kadapt. If "eigenvectors"-> True, returns each set of eigenvectors instead of each eigenvalue.

xflavor default has been set to true.

Ellipse fit now fits x neutrinos/anti neutrinos.

gitneutrinos committed:
typo

typo fixed in kAdapt. stability matrix was called with many radius data instead of single radius data.

gitneutrinos committed:
Update StabilityPackage.wl

removed some debugging print statements and debugging output.

gitneutrinos committed:
verification test needed option change

gitneutrinos committed:
updted .gitignore

gitneutrinos committed:
Merge branch 'xflavor' into master

srichers commented on StabilityPackage.wl:
```diff
@@ -211,18 +357,112 @@ Reap[
+getIntialGuess[data_,species_]:=Module[{ei,aei,ag,aag,\[Beta]g,a\[Beta]g,a\[Chi]g,\[Chi]g,g0,datasr,foc1234,afoc1234,xfoc1234,xag,xei},
+datasr=SelectSingleRadius[data,1];
+
+foc1234[x_,y_,z_,E_]:= ((3c^3)/(4 Pi h (1/2 (data["freq"][[E+1]]+data["freq"][[E]])) (data["freq"][[E+1]]^3-data["freq"][[E]]^3)) )(datasr["lotsodo"][[1,E,1]] 
```
would be better to combine these functions with a "species index" parameter and a "weight" parameter to prevent repeated code

gitneutrinos replied:
You mean combine the three "foc1234"s right?

srichers commented on StabilityPackage.wl:
```diff
@@ -211,18 +357,112 @@ Reap[
+
+Which[
+species==1,
+ne=Sum[ datasr["lotsodo"][[1,f,1]]/( h (1/2 (data["freq"][[f+1]]+data["freq"][[f]]))),{f,1,80}];
```
Again here, would be better to define a function that is passed a species index to prevent repeated code

srichers replied:
Ah, this function already is passed species. So here, you can put the weight inside of the which, and have the complicated expressions outside of the which and just use "weight" as a prefactor and "species" as the index

gitneutrinos replied:
Oh I see, got it.

Commit:
Delete RunStuff Template.nb

gitneutrinos committed:
Update .gitignore

gitneutrinos committed:
Revert "Update .gitignore"

This reverts commit 2c914d7ab4aa2b5cff000ab92a833dbe1e3ab195.

gitneutrinos committed:
Merge branch 'master' of https://github.com/gitneutrinos/stability_tools

gitneutrinos committed:
Reduced repeated code

Reduced repeated code in the ellipse fit section.  Included dispersion relation check. Include
